utilities/utility.py

```diff
@@ -14,6 +16,41 @@ def update_entity_fields(entity, **kwargs):
     return entity
 
 
+def cascade_soft_delete(parent_model, child, parent_id):
+    @event.listens_for(parent_model, 'after_update')
+    def receive_after_update(mapper, connection, target):
+        listen_for_after_flush_event(target, child, parent_id)
+
+
+def listen_for_after_flush_event(target, child, parent_id):
+    @event.listens_for(db_session, "after_flush", once=True)
+    def receive_after_flush(session, context):
+        """
+        Function to update related child data after a delete
+        mutation has been run.
+        The check statement on line 42 caters for the difference
+        in naming of room_resource folder and Resource class
+            :param session:
+            :param context:
+        """
+        if target.state == 'archived':
+            child_model = getattr(api, child)
+            child_object = getattr(
+                child_model.models,
+                'Resource' if child.capitalize() == 'Room_resource'
```

Comment:
Hello @RonKbS , good work. help me understand more about this code by explaining why there is a need to use the capitalize method here. Thanks

Reply:
@nakatuddesuzan, the method needs to run queries on an object like `api.room.models.Room` and to make it generic one string is passed in (`room`) that is then capitalized to `Room`